File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int rightMax = -1;

        for (int i = arr.size() - 1; i >= 0; i--) {
            int temp = arr[i];
            arr[i] = rightMax;
            rightMax = max(temp, rightMax);
        }
        return arr;
    }
};

File: Data Structures & Algorithms/valid-parenthesis-string/submission-1.py
class Solution:
    def checkValidString(self, s: str) -> bool:
        low, heigh = 0, 0

        for ch in s:
            if ch == "(":
                low += 1
                heigh += 1
            elif ch == ")":
                low = max(0, low - 1)
                heigh -= 1
            else:
                low = max(0, low - 1)
                heigh += 1

            if heigh < 0:
                return False

        return low == 0